src/back/wgsl/writer.rs

```diff
@@ -530,15 +671,79 @@ impl<W: Write> Writer<W> {
+                // WGSL does not support all SPIR-V builtins and we should skip it in generated shaders.
+                // We already skip them when we generate struct type.
+                // Now we need to find components that used struct with ignored builtins.
+
+                // So, why we can't just return the error to a user?
+                // We can, but otherwise, we can't generate WGSL shader from any glslang SPIR-V shaders.
+                // glslang generates gl_PerVertex struct with gl_CullDistance, gl_ClipDistance and gl_PointSize builtin inside by default.
+                // All of them are not supported by WGSL.
+
+                // We need to copy components to another vec because we don't know which of them we should write.
+                let mut components_to_write = Vec::with_capacity(components.len());
+                for component in components {
+                    let mut skip_component = false;
+                    if let Expression::Load { pointer } = func_ctx.expressions[*component] {
+                        let pointer_global = &func_ctx.info[pointer].assignable_global;
+                        if let Some(global_handle) = *pointer_global {
+                            // Let's ensure that we find the required global (gl_PerVertex).
+                            // spv-in generates a fake function for all outputs and global should be with WRITE usage
+                            if func_ctx.info[global_handle].contains(crate::valid::GlobalUse::WRITE)
+                            {
+                                if let Expression::AccessIndex {
+                                    base,
+                                    index: access_index,
+                                } = func_ctx.expressions[pointer]
+                                {
+                                    let base_ty_res = &func_ctx.info[base].ty;
+                                    let resolved = base_ty_res.inner_with(&module.types);
+                                    if let TypeInner::Pointer { base, class: _ } = *resolved {
+                                        // Let's check that we try to access a struct member with unsupported built-in and skip it.
+                                        if let TypeInner::Struct { ref members, .. } =
+                                            module.types[base].inner
+                                        {
+                                            for (member_index, member) in members.iter().enumerate()
+                                            {
+                                                if let Some(Binding::BuiltIn(builtin)) =
+                                                    member.binding
+                                                {
+                                                    if !is_builtin_supported(builtin)
+                                                        && member_index == access_index as usize
```

Comment:
if we know we are only looking for `member_index == access_index`, why not simply do `members[access_index as usize]` and work with it?